### PR TITLE
Scripts for running a dockerized local baseledger plateau network

### PR DESCRIPTION
**Pull request by ognjenkurtic**
This setup will be used to run integration and stress testing of the network

build-container.sh - Builds both the baseledgerd and orchestrator from the Dockerfile. Needs to be ran only once if no changes.

start-containers.sh - Fires up X number of validator containers on a single shared network with relevant exposed ports.

setup-validator.sh - Inits the validators, exchanges the genesis and gentx transactions among X number of validators.

run-testnet.sh - Fires up the baseledger node in each of the container with parameters for discoverability. Logs the output of each node to host/validatorX/logs.

clean.sh - Stops the network and cleans up everything.

### CODE REVIEW

File: baseledger/ops/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM starport/cli
```
Comment (skosito) [inline baseledger/ops/Dockerfile:1]: just to not forget, will this PR solve logging in containers?

File: tests/build-container.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -eux
+
+# this directy of this script
```
Comment (skosito) [inline tests/build-container.sh:4]: some typo here

File: tests/build-container.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+DOCKERFOLDER=$DIR/dockerfile
+REPOFOLDER=$DIR/..
+
+# change our directory so that the git archive command works as expected
+pushd $REPOFOLDER
+
+#docker system prune -a -f
```
Comment (skosito) [inline tests/build-container.sh:12]: this one should be removed i assume

File: tests/setup-validators.sh [hunk 4]
```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+set -eux
+# your baseledger binary name
+BIN=baseledgerd
+
+CHAIN_ID="baseledger"
+
+NODES=3
+
+ALLOCATION="10000000000stake,10000000000worktoken"
```
Comment (skosito) [inline tests/setup-validators.sh:10]: denom name is just work, not worktoken

File: tests/setup-validators.sh [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+# add in denom metadata for both native tokens
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.bank.denom_metadata += [{"name": "Work Token", "symbol": "WRK", "base": "worktoken", "description": "A test token for paying work", "denom_units": [{"denom": "worktoken", "exponent": 0}]},{"name": "Stake Token", "symbol": "STK", "base": "stake", "description": "A staking test token", "denom_units": [{"denom": "stake", "exponent": 0}]}]' /validator/config/genesis.json > /metadata-genesis.json
+docker cp /metadata-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/metadata-genesis.json
+
+# a 60 second voting period to allow us to pass governance proposals in the tests
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.gov.voting_params.voting_period = "60s"' /metadata-genesis.json > /edited-genesis.json
+docker cp /edited-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/edited-genesis.json
+
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME mv /edited-genesis.json /genesis.json
```
Comment (skosito) [inline tests/setup-validators.sh:32]: does this mean we can edit genesis here for more params changes (check BAS-140, we can turn off inflation here i assume? not relevant for tests i guess)

File: tests/setup-validators.sh [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+do
+# move the genesis in
+docker cp ./genesis.json $VALIDATOR_CONTAINER_BASE_NAME$i:/validator/config/genesis.json
+
+ARGS="$BASELEDGER_HOME --keyring-backend test"
+ORCHESTRATOR_KEY=$(docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys show orchestrator -a $ARGS)
+
+VALIDATOR_CONTAINER_IP=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $VALIDATOR_CONTAINER_BASE_NAME$i)
+
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN gentx $ARGS $BASELEDGER_HOME --moniker validator --chain-id=$CHAIN_ID --ip $VALIDATOR_CONTAINER_IP validator 500000000stake
```
Comment (skosito) [inline tests/setup-validators.sh:71]: small thing but should they have different monikers?

File: baseledger/ops/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+# override entrypoint from above image
+ENTRYPOINT []
+
+WORKDIR /apps
+
+USER root
+
+COPY --chmod=777 . .
+
+EXPOSE 26656 1317 26657
```
Comment (skosito) [inline baseledger/ops/Dockerfile:12]: should we expose 9090 and 26655?

**Review by skosito — APPROVED** [review]

File: baseledger/ops/Dockerfile (hunk 1 above)
Comment (ognjenkurtic) [reply baseledger/ops/Dockerfile:1]: Yes

File: tests/setup-validators.sh (hunk 5 above)
Comment (ognjenkurtic) [reply tests/setup-validators.sh:32]: Yes

File: tests/setup-validators.sh (hunk 6 above)
Comment (ognjenkurtic) [reply tests/setup-validators.sh:71]: actually not sure, i thought moniker is local, we can change

**Pull request merged**